### PR TITLE
neural network additions

### PR DESCRIPTION
**Pull request by kirsten-stallings**

### CODE REVIEW

File: ML_tutorial.ipynb [hunk 1]
```diff
@@ -6382,7 +6578,92 @@
...
+    "\n",
+    "No, the end is not near. Like any other machine learning algorithm, neural nets can get very, very, good at specific tasks. However, they cannot \"reason\". They cannot branch out beyond what they are trained to do. They are also only as good as the data they are trained on. Therefore, there is little risk of a robot revolution any time soon."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Neural nets are a black box\n",
+    "In the words of the Black-Eyed Peas, \"that's so 2000 and late\". In recent years, researchers have developed a variety of techniques to expose what each hidden node is focusing on. For example, researchers can determine what a convolutional neural network is focusing on when classifying images (hint: neural nets love edges and boundaries) ![pic](figures/inside_cnn.png)"
```
Comment (jambran) [inline ML_tutorial.ipynb]: 😆 😆 😆

File: ML_tutorial.ipynb [hunk 2]
```diff
@@ -6382,7 +6578,92 @@
...
+    "There are a lot of misconceptions about neural nets. The term is often thrown around like \"AI\". We want to take a moment to address a few of them\n",
+    "#you have to build it yourself"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Neural nets will take over the world\n",
+    "![pic](figures/cat_as_dog.png)\n",
```
Comment (jambran) [inline ML_tutorial.ipynb]: I think you're missing this image? `git status` has been super helpful for me with this

File: ML_tutorial.ipynb [hunk 3]
```diff
@@ -6382,7 +6578,92 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Training our own neural net: multi layered perceptron\n",
+    "\n",
+    "We will use scikit learn to create a [multilayer perceptron](http://deeplearning.net/tutorial/mlp.html).\n",
+    "\n",
+    "The main hyperparameters we must decide are how many hidden layers and the size of each hidden layer. For scikit learn, you create a tuple, where the length of your tuple is the number of hidden layers and the integer at tuple\\[i\\] is the size of the ith hidden layer. For example, if `hidden_layers = (32, 16, 8, 2)`, my model would have 4 hidden layers of size 32, 16, 8, and 2. The size of your last hidden layer should equal the number of classes. Since we are doing a binary classification, the size of last hidden layer should be 2."
```
Comment (jambran) [inline ML_tutorial.ipynb]: could you format `tuple[i]` here?

File: ML_tutorial.ipynb [hunk 4]
```diff
@@ -6391,29 +6672,88 @@
...
-   "execution_count": 70,
+   "cell_type": "markdown",
    "metadata": {},
-   "outputs": [],
    "source": [
-    "#To reload\n",
-    "with open(\"machine_learning_vocab.pkl\", \"rb\") as file:\n",
-    "    ml_vocab=pickle.load(file)"
+    "## Homework\n",
+    "A random sample of dialogue has been held out by us. Using only python code and packages, refine/create your best possible model to predict if a piece of dialogue is spoken by Tyrion Lannister. The person with the best performing model wins a prize to be determined (probably lunch). There will "
```
Comment (jambran) [inline ML_tutorial.ipynb]: oh no, did it not save the last bit of the sentence here?

File: ML_tutorial.ipynb (hunk 2 above)
Comment (kirsten-stallings) [reply ML_tutorial.ipynb]: thank you!

**Pull request merged**